src/components/add-service-form/add-service-form.jsx

```diff
@@ -13,8 +13,86 @@ const AddServiceForm = ({ onClose }) => {
     const [image, setImage] = useState(null);
     const [availableTimes, setAvailableTimes] = useState([]);
     const [error, setError] = useState('');
+    const [googleCalEvents, setGoogleCalEvents] = useState([]);
+    const [bookings, setBookings] = useState([]);
+    const [offeredBookings, setOfferedBookings] = useState([]);
     const serviceTypes = ['Hair', 'Clothes', 'Food', 'Cosmetic']; // Replace with actual service types
 
+    useEffect(() => {
+        fetchGoogleCal();
+        fetchBookings();
+        fetchOfferedBookings();
+    }, []);
+
+    const fetchGoogleCal = async () => {
+        try {
+            const token = localStorage.getItem('token');
+            if (!token) {
+                navigate('/login');
+                return;
+            }
+            const response = await fetch(`${import.meta.env.VITE_BACKEND_ADDRESS}/api/google-calendar/events`, {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Bearer ${token}`,
+                },
+            });
+            if (!response.ok) {
+                throw new Error('Failed to fetch Google Calendar events');
+            }
+            const googleData = await response.json();
+            setGoogleCalEvents(googleData);
+        } catch (error) {
+            console.error(error);
+        }
+    }
```

Comment:
The 3 functions in this file have some code that is common and could be pulled out into a new function. You could have a function that looks like this
```
function makeRequest(
    $url, $errorMessage
): {
            const token = localStorage.getItem('token');
            if (!token) {
                navigate('/login');
                return;
            }
            const response = await fetch($url, {
                method: 'GET',
                headers: {
                    'Authorization': `Bearer ${token}`,
                },
            });
            if (!response.ok) {
                throw new Error($errorMessage);
            }
            return response;
}
```

Then you can have
```
const fetchGoogleCal = async () => {
        try {
            const response = await makeRequest(
                `${import.meta.env.VITE_BACKEND_ADDRESS}/api/google-calendar/events`
                'Failed to fetch Google Calendar events'
            )
            const googleData = await response.json();
            setGoogleCalEvents(googleData);
        } catch (error) {
            console.error(error);
        }
    }
```

Reply:
Thank you winston i'll work on doing that and also in other components with similar code